proto/all_test.go

```diff
@@ -363,6 +364,75 @@ func TestMarshalerEncoding(t *testing.T) {
 	}
 }
 
+// Ensure that Buffer.Marshal uses O(N) memory for N messages
+func TestBufferMarshalAllocs(t *testing.T) {
```

Comment:
I'm wary of adding a test that relies on runtime information. It seems that this can actually be tested in a simpler way:
https://play.golang.org/p/r0PtrqqYxnk